lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -548,6 +547,54 @@ impl<T: FormatName> FormatListPDU<T> {
     }
 }
 
+// encode_clipboard encodes data suitably for clipboard storage.
+// This means determining the appropriate format and making sure the data has a nul terminator.
+//
+// This data must be valid UTF-8.
+fn encode_clipboard(mut data: Vec<u8>) -> (Vec<u8>, ClipboardFormat) {
+    if data.is_ascii() {
+        if matches!(data.last(), Some(x) if *x != b'\0') {
+            data.push(b'\0');
+        }
+
+        (data, ClipboardFormat::CF_TEXT)
+    } else {
+        let str = std::str::from_utf8(&data).unwrap();
```

Comment:
I have to say, not a big fan of this safely accessible panic here 😬

Are you sure there's no clean way to do the LF->CRLF translation while still staying in the comfy realm of `String`/`str`?

Reply:
I've now made this conversion without going to bytes and got rid of the unwrap.